### PR TITLE
Refactor and fix super issues

### PR DESCRIPTION
**Pull request by GarethDaviesGW**
some codeclimate issues with super and refactoring hdf.py to avoid repeated code in statmapdata and multiifostatmapdata classes

### CODE REVIEW

File: pycbc/io/hdf.py [hunk 1]
```diff
@@ -168,13 +168,26 @@ def remove(self, idx):
...
+
 
 class StatmapData(DictArray):
-    def __init__(self, data=None, seg=None, attrs=None,
-                       files=None):
-        groups = ['stat', 'time1', 'time2', 'trigger_id1', 'trigger_id2',
-               'template_id', 'decimation_factor', 'timeslide_id']
-        super(StatmapData, self).__init__(data=data, files=files, groups=groups)
+    def __init__(self, data=None, seg=None, attrs=None, files=None,
+                 groups=['stat', 'time1', 'time2', 'trigger_id1',
```
Comment (tdent) [inline pycbc/io/hdf.py]: Codeclimate doesn't seem to like this default assignment.  Can we instead set groups to None as default and assign it to the right list of strings just before calling super.__init__ ?

Comment (spxiwh) [reply pycbc/io/hdf.py]: If `groups` never changes (is immutable), then the right solution might be to make it a tuple, not a list?

Comment (GarethDaviesGW) [reply pycbc/io/hdf.py]: `groups ` **here** doesn't change, but the `groups` it is set as in MultiifoStatmapData does; it is appended to on lines 228-229. The workaround for which would be a bit messy I think

Comment (spxiwh) [reply pycbc/io/hdf.py]: You could use tuples also in MultiifoStatmapData, I think.

Comment (tdent) [reply pycbc/io/hdf.py]: Although the default value of `groups` can be a tuple I think we should be able to override it by other (iterable) type .. no?

Comment (GarethDaviesGW) [reply pycbc/io/hdf.py]: If I change both then it works

Comment (tdent) [reply pycbc/io/hdf.py]: @GarethDaviesGW we usually use python lists for lists of named parameters so I'd prefer to stick with lists except where they are deprecated, ie here as a default kwarg value.

Comment (GarethDaviesGW) [reply pycbc/io/hdf.py]: Tom's suggestion implemented in b664cc0 - tested and works with tuple in kwargs and list overwriting this in MultiifoStatmapData

**Comment by tdent** [comment]
Thanks @GarethDaviesGW.  I think the next step is to do command-line tests of both `pycbc_coinc_statmap` on coinc files from the cvmfs chunk22 production run and `pycbc_multiifo_coinc_statmap` on all the sets of coinc files from your chunk22 test run (the one done with the changes Marton's in previous commit) - check in either case that the content of the output files doesn't change in either case.  If that looks to be the case then add a comment here and we can merge.

**Pull request merged**